docker/data-science/notebooks/aips.py

```diff
@@ -117,7 +117,7 @@ def render_search_results(query, results):
 
         return rendered
 
-def create_view(collection, view_name, spark=None):
+def spark_view_from_collection(collection, view_name, spark=None):
```

Comment:
I think `sql_view_from_collection` or `spark_sql_view_from_collection`. I like the former if you think it's clear enough what it is, since the only SQL queries we do in the book are through Spark.

I don't think views are general Spark objects (like dataframes). I think a "view" is simply creating a SQL view from a dataframe fo use in SQL.

Reply:
That's right, missed that. I think create_table_from_view is most appropriate then

Reply:
Just to note, a "table" and a "view" have different meanings, though, A table is an actual materialized structure that is stored, and a view is a non-materialized representation (a query on other data) that can be queried.

In this case, these are views that can query into a dataframe, not literal tables being created. 

So I think SQL view is the right semantics that represent what it is.

https://www.geeksforgeeks.org/difference-between-view-and-table/

Reply:
Yes, they are different, but for the purposes of the reader and simplicity I leaned toward removing the implementation detail of what's being used in spark. For all our use cases it makes no difference to the consumer whether it's a view or a table. 

Also, this is a view in spark, this is not a SQL database. We're treating it like one, and again for the consumer the only thing that they care about is that SQL's going in and they're interacting w/ spark like it's a SQL DB. 

It's a matter of if we want to be explicit about what's happening. 

I think calling it a view is correct, but I don't think qualifying it as spark or spark_sql or sql (not actually sql) is necessary

Reply:
>I think calling it a view is correct, but I don't think qualifying it as spark or spark_sql or sql (not actually sql) is necessary

I agree. Let's call it a view and not a table. The word view is generic outside the context of a SQL query, which is why I thought prefixing with "sql_view" clarified this means "view" for a SQL query, not some other kind of view (model-view-controller, etc.) But with the `_from_collection` now added, I think this ambiguity about the meaning of "view" is mostly mitigates.

So **`create_view_from_collection` is fine to me if you don't want "sql" in the name.**